docs/content/docs/deployment/memory/network_mem_tuning.md

```diff
@@ -0,0 +1,159 @@
+
+Buffer debloating in Flink works by measuring past throguhput to predict future time to consume the remaining
+in-flight data. If those predictions are incorrect, the debloating mechanism can fail in one of the two ways:
+* There won't be enough buffered data to provide full throughput
+* There will be too many buffered in-flight data and the aligned checkpoint barriers propagation time or the unaligned checkpoint size will suffer.
+
+Hence, if you have a varying load in your job, for example a sudden spikes of incoming records, or periodically
+firing windowed aggregations or joins, you might need to adjust the following settings:
+
+* `taskmanager.network.memory.buffer-debloat.period` - The minimum time between buffer size recalculation.
+The shorter the period, the faster reaction time of the debloating mechanism, but a higher CPU overhead for the necessary calculations.
+* `taskmanager.network.memory.buffer-debloat.samples` - Adjust the number of samples over which throughput measurements are averaged out.
+The frequency of the collected samples can be adjusted via `taskmanager.network.memory.buffer-debloat.period`.
+The fewer samples, the faster reaction time of the debloating mechanism, but a higher chance of a sudden spike or drop of the throughput to cause the buffer debloating to miscalculate the best amount of the in-flight data.
+* `taskmanager.network.memory.buffer-debloat.threshold-percentages` - The optimization which prevents 
+the frequent buffer size change if the new size is not so different compared to the old one.
+
+See the [Configuration]({{< ref "docs/deployment/config" >}}#full-taskmanageroptions) documentation for details and additional parameters.
+
+The [metrics]({{< ref "docs/ops/metrics" >}}#io) which can help to observe the current buffer size:
+* `estimatedTimeToConsumerBuffersMs` - the total time to consume data from all input channels.
+* `debloatedBufferSize` - the current buffer size.
+
+## Network buffer lifecycle
+Logically, Flink has several local buffer pools one for output stream and one for each input gate. 
+Each of that pools is limited to at most 
+```
+#channels * taskmanager.network.memory.buffers-per-channel + taskmanager.network.memory.floating-buffers-per-gate
+```
+
+The size of the buffer can be configured by setting `taskmanager.memory.segment-size`.
+
+### Input network buffers
+Buffers in the input channel are divided into exclusive and floating buffers.
+Flink attempts to acquire the configured amount of the exclusive buffers in the initialization phase for each channel.
+Exclusive buffers can be used only by one particular channel.
+A channel can request additional floating buffers from a buffer pool shared across all channels belonging to the given input gate.
+All exclusive buffers must be fulfilled on initialization, otherwise Flink will throw an exception 
+while the floating buffers are treated as recommended, a single available buffer will be enough for Flink to make a progress.
```

Comment:
> a single available buffer will be enough for Flink to make a progress.

What does it refer to? I presume only to floating buffers? If so, I would split his long sentence into to:

> All exclusive buffers must be fulfilled on initialization, otherwise job will fail with an exception. 
> On top of that single floating buffer has to be allocated during initialization for Flink to be able to make progress.
> Remaining floating buffers are optional and are acquired only if there are enough resource available.